iocore/net/SSLUtils.cc

```diff
@@ -1842,6 +1842,16 @@ SSLInitServerContext(const SSLConfigParams *params, const ssl_user_config *sslMu
       goto fail;
     }
   }
+
+#if (OPENSSL_VERSION_NUMBER >= 0x10101000L)
```

Comment:
Is there any other ifdef or detection we can use? Or do we know that this will be for OpenSSL only, and never BroimgSSL etc?

Reply:
Another choice is adding macros to detect this API in `/build/crypto.m4`. Do you prefer that?

BoringSSL has this macro and it indicates API compatibility.
> BoringSSL's OPENSSL_VERSION_NUMBER matches the OpenSSL version it targets. Version checks for OpenSSL should ideally work as-is in BoringSSL

Reply:
ifdef SSL_CTX_set1_groups_list ?

Reply:
If you're willing to do a bit of meta programming, I have a header file with which you could do this
```
namespace
{
template <typename T>
auto
TS_SSL_CTX_set1_groups_list(T *, char *, ts::meta::CaseArg_0) -> bool
{
  return true;
}

template <typename T>
auto
TS_SSL_CTX_set1_groups_list(T *ctx, char *groups, ts::meta::CaseArg_1)
  -> decltype(SSL_CTX_set1_groups_list(static_cast<T *>(nullptr), static_cast<char*>(nullptr)), bool())
{
  if (!SSL_CTX_set1_groups_list(ctx, groups)) {
    SSLError("invalid groups list for client in records.config");
    return false;
  }
  return true;
}
} // namespace

bool
TS_SSL_CTX_set1_groups_list(SSL_CTX *ctx, char *params)
{
  return TS_SSL_CTX_set1_groups_list(ctx, params, ts::meta::Case_Arg);
}
```
At the original call site, it's
```
  if (params->client_groups_list != nullptr) {
    if (!TS_SSL_CTX_set1_groups_list(client_ctx, params->client_groups_list)) {
      SSLError("invalid groups list for client in records.config");
      goto fail;
    }
  }
```
This doesn't depend on any `#define` or checking in `configure.ac`, it's completely self contained. It doesn't depend on knowing the openSSL version, and should work on BoringSSL. In fact, if BoringSSL has a functionally equivalent but differently named function, that could be easily incorporated.

Reply:
Interesting! Could you open a new PR to add the magic header file? It looks we should discuss ifdef v.s. meta programing for checking existence of libraries APIs. Because we have bunch of checks like this.